### ISSUE TITLE
[bug] RoFormer seq/token-classification heads return MaskedLMOutput, breaking the by-type output contract

### ISSUE THREAD

**Issue by ChanLumerico** [ChanLumerico/lucid#75]
## Summary

Two RoFormer heads declare and return `MaskedLMOutput` where they should return the semantically-correct output dataclass:

- `lucid/models/text/roformer/_model.py:698` — `RoFormerForSequenceClassification.forward` is annotated `-> MaskedLMOutput` and returns `MaskedLMOutput(logits=logits, loss=loss)` at `lucid/models/text/roformer/_model.py:717`. It should return **`SequenceClassificationOutput`**.
- `lucid/models/text/roformer/_model.py:798` — `RoFormerForTokenClassification.forward` is annotated `-> MaskedLMOutput` and returns `MaskedLMOutput(logits=logits, loss=loss)` at `lucid/models/text/roformer/_model.py:814`. It should return **`TokenClassificationOutput`**.

**Bug class:** wrong return type / broken output contract (BERT-parity regression).

## Why this is a bug

This is the same defect the recent commit `43fa1f9` ("the span head returns what BERT's returns") fixed for `RoFormerForQuestionAnswering` — its own message noted "the same mistake twice in one file suggests the output types were picked by copying a neighbour." Two more instances of that mistake remain in the file.

The framework's design makes the output *type* load-bearing, not just its fields:

- `SequenceClassificationOutput` docstring (`lucid/models/_output.py:678`): *"It exists so a caller can tell a `(B, num_labels)` sentence logit apart from a `(B, T, vocab_size)` masked-LM logit **by type** rather than by inspecting shapes."*
- `TokenClassificationOutput` docstring (`lucid/models/_output.py:715`): *"the distinct type keeps the two from being confused."*
- `lucid/models/_tasks.py`: `SequenceClassificationModel` "Returns `SequenceClassificationOutput`"; `TokenClassificationModel` "Returns `TokenClassificationOutput`". Both RoFormer classes subclass those task bases, so they violate their own base's promise.

BERT's analogous heads are correct: `BERTForSequenceClassification.forward -> SequenceClassificationOutput` (`lucid/models/text/bert/_model.py:831`) and `BERTForTokenClassification.forward -> TokenClassificationOutput` (`lucid/models/text/bert/_model.py:931`).

**Concrete failure case:**
```python
from lucid.models import SequenceClassificationOutput
out = roformer_seq_cls_model(input_ids)
isinstance(out, SequenceClassificationOutput)  # False for RoFormer, True for BERT
```
Any consumer that dispatches on output type (or trusts the declared return annotation under `mypy --strict`) misclassifies RoFormer's sequence- and token-classification results, even though the identical BERT head works. The four dataclasses currently share the same fields (`logits`, `loss`, `hidden_states`, `attentions`), so this does not crash today — which is exactly why it is easy to miss — but it defeats the by-type discrimination the output types were built for.

## Suggested fix

Mirror the `43fa1f9` QA fix for the other two heads:

1. Import `SequenceClassificationOutput` (already imported) and `TokenClassificationOutput` in `lucid/models/text/roformer/_model.py`.
2. `RoFormerForSequenceClassification.forward`: change the annotation to `-> SequenceClassificationOutput` and return `SequenceClassificationOutput(logits=logits, loss=loss)`.
3. `RoFormerForTokenClassification.forward`: change the annotation to `-> TokenClassificationOutput` and return `TokenClassificationOutput(logits=logits, loss=loss)`.

Add/adjust the doctest and any output-type assertions accordingly, as the QA fix did.
